File: platforms/hacker-earth/E--minimum-length-4-945227e2/solutions/solution.py
def solve(n, a, b):
    i, j = 0, n - 1
    while i <= j:
        if a[i] == b[i]:
            i += 1
        elif a[j] == b[j]:
            j -= 1
        else:
            break
    return j - i + 1


t = int(input())
for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    print(solve(n, a, b))

File: platforms/hacker-earth/E--minimum-length-4-945227e2/solutions/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func solve(n int, a []int, b []int) int {
	i, j := 0, n-1
	for i <= j {
		if a[i] == b[i] {
			i++
		} else if a[j] == b[j] {
			j--
		} else {
			break
		}
	}
	return j - i + 1
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	t, _ := strconv.Atoi(scanner.Text())

	scanArr := func() []int {
		scanner.Scan()
		fields := strings.Fields(scanner.Text())
		arr := make([]int, 0)
		for _, field := range fields {
			val, _ := strconv.Atoi(field)
			arr = append(arr, val)
		}
		return arr
	}

	for ; t > 0; t-- {
		scanner.Scan()
		n, _ := strconv.Atoi(scanner.Text())

		a := scanArr()
		b := scanArr()
		fmt.Println(solve(n, a, b))
	}
}
